csrc/id_model/transform_replay.h

```diff
@@ -0,0 +1,103 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <c10/macros/Export.h>
+
+#include <ir/all_nodes.h>
+
+#include <unordered_map>
+#include <vector>
+
+namespace nvfuser {
+
+class ReplayTransform : OptInConstDispatch {
```

Comment:
Looks like this is pretty similar to `ReplaySelf`. I understand it's just a small class, but is there any specific reason not to reuse `ReplaySelf`?

Reply:
I don’t recall any specific reason.